mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -308,40 +326,54 @@ def evaluate_comparison(self, node: mparser.ComparisonNode) -> InterpreterObject
 
     def evaluate_andstatement(self, cur: mparser.AndNode) -> InterpreterObject:
         l = self.evaluate_statement(cur.left)
+        if l is None:
+            raise InvalidCodeOnVoid()
         if isinstance(l, Disabler):
             return l
         l_bool = l.operator_call(MesonOperator.BOOL, None)
         if not l_bool:
             return self._holderify(l_bool)
         r = self.evaluate_statement(cur.right)
+        if r is None:
+            raise InvalidCodeOnVoid()
         if isinstance(r, Disabler):
             return r
         return self._holderify(r.operator_call(MesonOperator.BOOL, None))
 
     def evaluate_orstatement(self, cur: mparser.OrNode) -> InterpreterObject:
         l = self.evaluate_statement(cur.left)
+        if l is None:
+            raise InvalidCodeOnVoid()
         if isinstance(l, Disabler):
             return l
         l_bool = l.operator_call(MesonOperator.BOOL, None)
         if l_bool:
             return self._holderify(l_bool)
         r = self.evaluate_statement(cur.right)
+        if r is None:
+            raise InvalidCodeOnVoid()
         if isinstance(r, Disabler):
             return r
         return self._holderify(r.operator_call(MesonOperator.BOOL, None))
 
     def evaluate_uminusstatement(self, cur: mparser.UMinusNode) -> InterpreterObject:
         v = self.evaluate_statement(cur.value)
+        if v is None:
+            raise InvalidCodeOnVoid()
         if isinstance(v, Disabler):
             return v
         v.current_node = cur
         return self._holderify(v.operator_call(MesonOperator.UMINUS, None))
 
     def evaluate_arithmeticstatement(self, cur: mparser.ArithmeticNode) -> InterpreterObject:
         l = self.evaluate_statement(cur.left)
+        if l is None:
+            raise InvalidCodeOnVoid()
         if isinstance(l, Disabler):
             return l
         r = self.evaluate_statement(cur.right)
+        if r is None:
+            raise InvalidCodeOnVoid()
```

Comment:
This actually is needed, because we can't evaluate l to see what its operator calling convention is.

r is actually sort of kind of okay-ish, we unholder it and that fails because "Unknown object None of type NoneType in the parameters.", but we claim this is a MesonBug, so that's an improvement here.